docs/en/stack/ml/nlp/ml-nlp-elastic-rerank.asciidoc

```diff
@@ -0,0 +1,356 @@
+// https://ml-models.elastic.co/rerank-v1_linux-x86_64.vocab.json
+// ```
+
+[discrete]
+=== Using an HTTP server
+
+INFO: If you use an existing HTTP server, note that the model downloader only 
+supports passwordless HTTP servers.
+
+You can use any HTTP service to deploy the model. This example uses the official 
+Nginx Docker image to set a new HTTP download service up.
+
+. Download the <<ml-nlp-rerank-model-artifacts,model artifact files>>.
+. Put the files into a subdirectory of your choice.
+. Run the following commands:
++
+--
+[source, shell]
+--------------------------------------------------
+export ELASTIC_ML_MODELS="/path/to/models"
+docker run --rm -d -p 8080:80 --name ml-models -v ${ELASTIC_ML_MODELS}:/usr/share/nginx/html nginx
+--------------------------------------------------
+
+Don't forget to change `/path/to/models` to the path of the subdirectory where 
+the model artifact files are located.
+
+These commands start a local Docker image with an Nginx server with the 
+subdirectory containing the model files. As the Docker image has to be 
+downloaded and built, the first start might take a longer period of time. 
+Subsequent runs start quicker.
+--
+. Verify that Nginx runs properly by visiting the following URL in your 
+browser:
++
+--
+```
+http://{IP_ADDRESS_OR_HOSTNAME}:8080/rerank-v1.metadata.json
+```
+
+If Nginx runs properly, you see the content of the metdata file of the model.
+--
+. Point your {es} deployment to the model artifacts on the HTTP server
+by adding the following line to the `config/elasticsearch.yml` file: 
++
+--
+```
+xpack.ml.model_repository: http://{IP_ADDRESS_OR_HOSTNAME}:8080
+```
+
+If you use your own HTTP or HTTPS server, change the address accordingly. It is 
+important to specificy the protocol ("http://" or "https://"). Ensure that all 
+master-eligible nodes can reach the server you specify.
+--
+. Repeat step 5 on all master-eligible nodes.
+. {ref}/restart-cluster.html#restart-cluster-rolling[Restart] the 
+master-eligible nodes one by one.
+. Create an inference endpoint to deploy the model per <<ml-nlp-rerank-deploy-steps,these steps>>.
+
+The HTTP server is only required for downloading the model. After the download 
+has finished, you can stop and delete the service. You can stop the Docker image 
+used in this example by running the following command:
+
+[source, shell]
+--------------------------------------------------
+docker stop ml-models
+--------------------------------------------------
+
+[discrete]
+=== Using file-based access
+
+For a file-based access, follow these steps:
+
+. Download the <<ml-nlp-rerank-model-artifacts,model artifact files>>. 
+. Put the files into a `models` subdirectory inside the `config` directory of 
+your {es} deployment.
+. Point your {es} deployment to the model directory by adding the 
+following line to the `config/elasticsearch.yml` file:
++
+--
+```
+xpack.ml.model_repository: file://${path.home}/config/models/`
```

Comment:
theres an extra backtick at the end of this line

Reply:
nice catch, fixing!